presto-main/src/main/java/io/prestosql/execution/SqlTaskManager.java

```diff
@@ -373,21 +373,22 @@ public TaskInfo updateTask(Session session, TaskId taskId, Optional<PlanFragment
         requireNonNull(sources, "sources is null");
         requireNonNull(outputBuffers, "outputBuffers is null");
 
-        long sessionQueryMaxMemoryPerNode = getQueryMaxMemoryPerNode(session).toBytes();
-        long sessionQueryTotalMaxMemoryPerNode = getQueryMaxTotalMemoryPerNode(session).toBytes();
-        // Session property query_max_memory_per_node is used to only decrease memory limit
-        if (sessionQueryMaxMemoryPerNode <= queryMaxMemoryPerNode) {
-            queryContexts.getUnchecked(taskId.getQueryId()).setMaxUserMemory(sessionQueryMaxMemoryPerNode);
```

Comment:
We know we can skip calling `setMaxUserMemory` because it will be obliterated by `setResourceOvercommit` invoc if overcommit enabled.
i.e. This is correct because of what `setResourceOvercommit` does internally.

I'd suggest removing (inlining) the `setResourceOvercommit` method, and exposing this relationship here:
```
if (no overcocommit) { 
 do usual stuff
}
else {
 setMaxUserMemory & setMaxTotalMemory to memoryPool.max
}
```

Reply:
@findepi I thought about this, and feel that it still may not be very clean, since we end up locking the object outside of the class, because we'd like to also access the guarded memoryPool. do you feel that the following would look better?

```
......
else {
    synchronized (queryContext) {
        long maxBytes = queryContext.getMemoryPool().getMaxBytes();
        queryContext.setMaxUserMemory(maxBytes);
        queryContext.setMaxTotalMemory(maxBytes);
    }
}
```

 Another option is we introduce specific methods like `setMaxUserMemoryToMemoryPoolMaxBytes` `setMaxTotalMemoryToMemoryPoolMaxBytes`. Thoughts?

Reply:
I see your point. Let's get in what you have now.